determined-ai/determined
# feat: cli: support downloading checkpoints through master

## Description

The CLI supports downloading checkpoints through the master. This is a follow-up PR for change #4989 and depends on the API added there.




## Test Plan



- Use `devcluster` to start a dev cluster. Configure the `checkpoint_storage` section to use an S3 bucket and provide appropriate AWS credentials. For example:
```yaml
checkpoint_storage:
  type: s3
  bucket: hanyu-det-test
  prefix: /devcluster/checkpoint-download-pr
  access_key: *** (redacted) ***
  secret_key: *** (redacted) ***
```
- Make sure the same or equivalent credentials are available in environment variables or `~/.aws/credentials.
- Run `export DET_MASTER=http://localhost:8080`
- Run a simple experiment, e.g., the MNIST example. When the experiment completes, a checkpoint should have been created.
- Obtain the checkpoint UUID under that experiment using the CLI.
- Go to the S3 bucket in the AWS Console to verify the checkpoint with the same UUID.
- Substitute `$UUID` in the following steps:
  1. Remove any previously downloaded checkpoints (if any)
  1. Run: `det c download --mode=master $UUID`
  1. Verify the contents of the downloaded checkpoint are the same as those on S3.
  1. Repeat steps i, ii, and iii above using these two additional commands:
    - `det c download --mode=direct $UUID`
    - `det c download --mode=auto $UUID`

## Commentary (optional)




## Checklist
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.





[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
